### PR TITLE
feat(1329): System table for system variables

### LINKED ISSUE #1329: Keep SQL variables in a system table
Right now we have environment variables that are runtime configurable through the SQL shell. These variables control database state but are not written to the commitlog as of yet. We should store them in a system table in order to track their state properly.

### PR DESCRIPTION
**Pull request by joshua-spacetime**
Closes #1329.

Used to persist global parameters for slow query logging and cardinality limits.

# Expected complexity level and risk

2

# Testing

*Describe any testing you've done, and any testing you'd like your reviewers to do,
so that you're confident that all the changes work as expected!*

- [x] *Write a test you've completed here.*
- [ ] *Write a test you want a reviewer to do here, so they can check it off when they're satisfied.*

### CODE REVIEW

File: crates/core/src/db/datastore/system_tables.rs [hunk 1]
```diff
@@ -716,10 +751,419 @@ impl From<&StClientsRow> for ProductValue {
...
+            | StVarName::SlowQueryThreshold
+            | StVarName::SlowSubThreshold
+            | StVarName::SlowIncThreshold => StVarType::U64,
+        }
+    }
+}
+
+/// The type of a system variable in `st_var`
+#[repr(u8)]
+pub enum StVarType {
```
Comment (mamcx) [inline crates/core/src/db/datastore/system_tables.rs:901]: Why do we need another `type` for this? I think just matching on `AlgebraicType` if wanna restrict to `primitive` types is enough

Comment (joshua-spacetime) [reply crates/core/src/db/datastore/system_tables.rs:901]: Can a column be typed as `AlgebraicValue`? I was under the assumption it needed a concrete type. In this case, `AlgebraicType::Sum(...)`.

Comment (gefjon) [reply crates/core/src/db/datastore/system_tables.rs:901]: We could conceivably `impl SpacetimeType for AlgebraicValue` by having an `AlgebraicType` which describes `AlgebraicValue`.

File: crates/core/src/vm.rs [hunk 2]
```diff
@@ -402,13 +405,14 @@ pub struct DbProgram<'db, 'tx> {
...
+    ctx: &ExecutionContext,
+    db: &RelationalDB,
+    tx: &TxId,
+    row_est: impl Fn(&QuerySet, &TxId) -> u64,
     auth: &AuthCtx,
-    config: &DatabaseConfig,
 ) -> Result<(), DBError> {
     if auth.caller != auth.owner {
-        if let Some(limit) = config.row_limit {
+        if let Some(limit) = StVarTable::row_limit(ctx, db, tx)? {
```
Comment (joshua-spacetime) [inline crates/core/src/vm.rs:415]: We use the `row_limit` in `st_var` when estimating cardinality.

**Review by mamcx — APPROVED** [review]
LGTM

**Pull request merged**